src/DropdownTypeaheadReferenceSet/components/DropdownTypeaheadReferenceSet.ts

```diff
@@ -0,0 +1,137 @@
+import { Component, createElement } from "react";
+import Select , { Async } from "react-select";
+import * as classNames from "classnames";
+
+import { Alert } from "./Alert";
+import { Label } from "./Label";
+
+import "../ui/DropdownTypeaheadReferenceSet.scss";
+import "react-select/dist/react-select.css";
+
+export interface DropdownTypeaheadProps {
+    styleObject?: object;
+    labelWidth: number;
+    data?: ReferenceOption[];
+    asyncData?: any;
+    labelCaption: string;
+    loaded: boolean;
+    showLabel: boolean;
+    emptyOptionCaption: string;
+    isClearable: boolean;
+    isReadOnly: boolean;
+    selectType: "normal" | "asynchronous";
+    selectedValue: any;
+    handleOnchange?: (selectedOption: ReferenceOption | any) => void;
+    className?: string;
+    readOnlyStyle: "control" | "text";
+    labelOrientation: "horizontal" | "vertical";
+    alertMessage: string;
+}
+
+export interface ReferenceOption {
+    value?: string | boolean;
+    label?: string;
+}
+
+export interface AttributeType {
+    name: string;
+    sort: string;
+}
+
+export class DropdownTypeahead extends Component<DropdownTypeaheadProps> {
+    render() {
+        return this.renderForm();
+    }
+
+    private renderForm() {
+        if (!this.props.loaded) {
+            if (this.props.showLabel && this.props.labelCaption) {
+                return createElement(Label, {
+                    className: this.props.className,
+                    label: this.props.labelCaption,
+                    orientation: this.props.labelOrientation,
+                    style: this.props.styleObject,
+                    weight: this.props.labelWidth
+                }, this.renderSelector());
+            }
+
+            return this.renderSelector();
+        }
+
+        return createElement("div", {});
+    }
+
+    private renderSelector() {
+        const commonProps: object = {
+            clearable: this.props.isClearable,
+            multi: true,
+            removeSelected: true,
+            disabled: this.props.isReadOnly,
+            onChange: this.props.handleOnchange,
+            clearValueText: "",
+            ...this.createSelectorProp() as object
+        };
+
+        if (this.props.readOnlyStyle === "control") {
+                return createElement("div", {
+                    className: classNames("widget-dropdown-type-ahead-wrapper")
+                },
+                this.props.selectType === "normal" ?
+                    this.props.isReadOnly ?
+                    createElement("input", {
+                        type: "text",
+                        readonly: "readonly",
+                        className: "form-control",
+                        disabled: "disabled",
+                        value: this.processOptions() }) :
+                    createElement(Select, {
+                        options: this.props.data,
+                        noResultsText: "No options",
```

Comment:
Should be support this? if so it should be translatable String
For now make text empty
```suggestion
                        noResultsText: "",
```

Reply:
Done